prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -361,6 +377,81 @@ ssize_t rxr_pkt_post_ctrl_or_queue(struct rxr_ep *ep, int entry_type, void *x_en
 	return err;
 }
 
+/*
+ * This function should be called before sending a message with the new protocol
+ * for FI_DELIVERY_COMPLETE.
+ *
+ * ep: The endpoint on which the packet for triggering handshake will be sent.
+ * peer: The peer from which the sender receives handshake.
+ * source_tx_entry: The tx_entry for the trigger packet.
+ *
+ * This function will return 0 if sender successfully received the handshake from the peer
+ *
+ * This function will return FI_EAGAIN if it fails to allocate or send the trigger packet or
+ * it fails to receive handshake packet within a certain period of time.
+ */
+
+ssize_t rxr_pkt_wait_handshake(struct rxr_ep *ep, struct rxr_tx_entry *source_tx_entry, struct rxr_peer *peer)
+{
+	struct rxr_tx_entry *tx_entry;
+	ssize_t err;
+
+	time_t start, endwait;
+	time_t timeout = 20; //in seconds
```

Comment:
20 seconds is way too much, I think 1,000,000 micros second is more than enough. You can call ofi_gettime_us() to get microsecond.